package/PartSeg/common_gui/napari_image_view.py

```diff
@@ -0,0 +1,552 @@
+
+    def has_image(self, image: Image):
+        return image.file_path in self.image_info
+
+    @staticmethod
+    def calculate_filter(array: np.ndarray, parameters: Tuple[NoiseFilterType, float]) -> np.ndarray:
+        if parameters[0] == NoiseFilterType.No or parameters[1] == 0:
+            return array
+        if parameters[0] == NoiseFilterType.Gauss:
+            return gaussian(array, parameters[1])
+        else:
+            return median(array, int(parameters[1]))
+
+    def add_image(self, image: Optional[Image]):
+        if image is None:
+            image = self.settings.image
+
+        if not image.channels:
+            raise ValueError("Need non empty image")
+
+        if image.file_path in self.image_info:
+            raise ValueError("Image already added")
+
+        channels = image.channels
+        if self.image_info:
+            channels = max(channels, *[x.image.channels for x in self.image_info.values()])
+
+        self.channel_control.set_channels(channels)
+        visibility = self.channel_control.channel_visibility
+        limits = self.channel_control.get_limits()
+        limits = [image.get_ranges()[i] if x is None else x for i, x in enumerate(limits)]
+        gamma = self.channel_control.get_gamma()
+        filters = self.channel_control.get_filter()
+        image_layers = []
+
+        for i in range(image.channels):
+            image_layers.append(
+                self.viewer.add_image(
+                    self.calculate_filter(image.get_channel(i), filters[i]),
+                    colormap=self.convert_to_vispy_colormap(self.channel_control.selected_colormaps[i]),
+                    visible=visibility[i],
+                    blending="additive",
+                    scale=image.normalized_scaling(),
+                    contrast_limits=limits[i],
+                    gamma=gamma[i],
+                )
+            )
+        if not self.image_info:
+            image_layers[0].blending = "translucent"
+        self.image_info[image.file_path] = ImageInfo(image, image_layers, filters)
+        self.current_image = image.file_path
+        if image.mask is not None:
+            self.set_mask()
+        return image
+
+    def images_bounds(self) -> Tuple[List[int], List[int]]:
+        ranges = []
+        for image_info in self.image_info.values():
+            if not image_info.layers:
+                continue
+            ranges = [
+                (min(a, b), max(c, d), min(e, f))
+                for (a, c, e), (b, d, f) in itertools.zip_longest(
+                    image_info.layers[0].dims.range, ranges, fillvalue=(np.inf, -np.inf, np.inf)
+                )
+            ]
+
+        visible = [ranges[i] for i in self.viewer.dims.displayed]
+        min_shape, max_shape, _ = zip(*visible)
+        size = np.subtract(max_shape, min_shape)
+        return size, min_shape
+
+    def _shift_layer(self, layer: Layer, translate_2d):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/Czaki/PartSeg/pullRequest?prid=5714517)